src/lbaf/Applications/LBAF_app.py

```diff
@@ -21,77 +21,139 @@
 
 # pylint: disable=C0413
 from lbaf.Utils.exception_handler import exc_handler
-from lbaf.Utils.colors import green
+from lbaf.Utils.logger import logger
 from lbaf.Utils.functions import abspath_from
 # pylint: enable=C0413
 
-def get_config_file() -> str:
-    """Parses command line argument and returns config file path."""
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--config", help="Path to the config file.", default='conf.yaml')
-    args = parser.parse_args()
-    if args.config:
-        # try to search the file from this place
-        config_file = os.path.abspath(args.config)
-        # if not found we might search in the config directory at project root
-        # but some thing will be disturbing : in config files we have path that must be relative
-        # to the current directory but then we might prefer paty to be relative to the configuration
-        # file location
-        if not os.path.isfile(config_file):
-            config_dir = os.path.join(
-                f"{os.sep}".join(os.path.abspath(__file__).split(os.sep)[:-4]), "config"
-            )
-            config_file = config_dir + '/' + args.config
-    else:
-        sys.excepthook = exc_handler
-        raise FileNotFoundError("Please provide path to the config file with '--config' argument.")
-
-    if not os.path.isfile(config_file):
-        raise FileNotFoundError(f"File not found at path {args.config}")
-    return config_file
-
-
-def check_and_get_schema_validator():
-    """Makes sure that SchemaValidator can be imported, and it's the latest version available."""
-    module_name = green(f"[{os.path.splitext(os.path.split(__file__)[-1])[0]}]")
-
-    def save_schema_validator_and_init_file(import_dir: str):
-        with open(os.path.join(import_dir, '__init__.py'), 'wt', encoding='utf-8') as init_file:
-            init_file.write('\n')
-        try:
-            script_name = "JSON_data_files_validator.py"
-            script_url = f"https://raw.githubusercontent.com/DARMA-tasking/vt/develop/scripts/{script_name}"
-            filename = urlretrieve(script_url, os.path.join(import_dir, script_name))
-            print(f"{module_name} Saved SchemaValidator to: {filename}")
-        except HTTPError as err:
+class Loader:
+    """Class to load LBAF configuration file and SchemaValidator script"""
+    def __resolve_config_file(self) -> str:
+        """Parses command line argument and resove the config file path."""
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--config",
```

Comment:
I suggest that we use a non-abbreviated option (`--configuration`) for the long form of the this flag, possibly with an added short form as well for convenience (`-c`).